scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,223 @@
+        raise Exception(
+            'File at path {} doesn\'t exist'.format(PR_LCOV_FILE_PATH))
+
+    tests = get_lcov_file_tests(PR_LCOV_FILE_PATH)
+    for lines in tests:
+        total_lines = re.match('(\d+)', lines[1].split(':')[1]).group(1)
+        covered_lines = re.match('(\d+)', lines[2].split(':')[1]).group(1)
+
+        test_name = get_test_file_name(lines[0])
+        coverage_dict[test_name] = [
+            int(total_lines),
+            int(covered_lines)
+        ]
+
+    return coverage_dict        
+
+
+def remove_item_from_whitelist(test_name):
+    index = FULLY_COVERED_TESTS.index(test_name)
+    FULLY_COVERED_TESTS.pop(index)
+
+
+def check_coverage_changes():
+    """Checks if the whitelist for fully covered files needs to be changed by:
+    - New file insertion
+    - File renaming
+    - File deletion
+
+    Raises:
+        Exception: If PR_LCOV_FILE_PATH doesn't exist.
+    """
+    if not os.path.exists(PR_LCOV_FILE_PATH):
+        raise Exception('File at path {} doesn\'t exist'.format(
+            PR_LCOV_FILE_PATH))
+
+    covered_tests = get_coverage_dict_tests()
+    errors = ''
+
+    for test_name in covered_tests:
+        total_lines = covered_tests[test_name][0]
+        covered_lines = covered_tests[test_name][1]
+
+        if test_name in FULLY_COVERED_TESTS:
+            if total_lines != covered_lines:
+                errors += ('\033[1m{}\033[0m file is in the whitelist but its'
+                ' coverage decreased. Make sure it is fully coverage'
+                ' again.\n'.format(test_name))
+
+            remove_item_from_whitelist(test_name)
+        else:
+            if total_lines == covered_lines:
+                errors += ('\033[1m{}\033[0m file is fully coverage and it\'s'
+                ' not included in the whitelist. Please add the file name in'
+                ' the whilelist manually.\n'.format(test_name))
+
+    if len(FULLY_COVERED_TESTS) > 0:
+        for test_name in FULLY_COVERED_TESTS:
+            errors += ('\033[1m{}\033[0m is in the whitelist but it doesn\'t'
+            ' exist anymore. If you have renamed it, please make sure to'
+            ' remove the old file name and add the new file name in the'
+            ' whitelist.\n'.format(test_name))
+
+    delete_tmp_folder()
+
+    if errors:
+        python_utils.PRINT('------------------------------------')
+        python_utils.PRINT('Frontend Coverage Checks Not Passed.')
+        python_utils.PRINT('------------------------------------')
+        python_utils.PRINT(errors)
+        sys.exit(1)
+    else:
+        python_utils.PRINT('------------------------------------')
+        python_utils.PRINT('All Frontend Coverage Checks Passed.')
+        python_utils.PRINT('------------------------------------')
+
+
+def main():
+    """Runs all the steps for checking if there is any decrease of 100% covered
+    files. Only PR branches is going to be checked, develop branch doesn't need
```

Comment:
Only PR branches will be checked. The develop branch doesn't ...

Reply:
Done.